### PR TITLE
Feature/sort planting

### PR DESCRIPTION
**Pull request by Adamws33**
added a state drop down that will update with map selection.  Also refactored landing component

### CODE REVIEW

**Comment by mikahpinegar** [comment]
we want to be able to sort the calendar view also

**Comment by mikahpinegar** [comment]
took me  a while to figure out the months are clickable. i think there should be a button called `Sort by Planting Window` somewhere so it's more obvious

File: src/pages/Landing/Landing.js [hunk 1]
```diff
@@ -131,14 +164,56 @@ const Landing = ({ height, title, bg }) => {
...
+                {st.label?.toUpperCase()}
+              </MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+      </ListItem>
+    </List>
+  );
+
+  useEffect(() => {
```
Comment (mikahpinegar) [inline src/pages/Landing/Landing.js:203]: there are two useEffects with the same dependencies

File: src/pages/Landing/Landing.js [hunk 2]
```diff
@@ -131,14 +164,56 @@ const Landing = ({ height, title, bg }) => {
...
+    if (selectedState) {
+      stateSelect();
```
Comment (mikahpinegar) [inline src/pages/Landing/Landing.js:205]: why are you calling a component in this useEffect?

File: src/pages/Landing/Landing.js (hunk 1 above)
Comment (mikahpinegar) [reply src/pages/Landing/Landing.js:203]: pretty sure this whole useEffect can be deleted

**Review by mikahpinegar — CHANGES_REQUESTED** [review]

File: src/shared/constants.js [hunk 3]
```diff
@@ -533,6 +533,31 @@ export const sortCrops = (type = 'Average Goals', crops = [], sortFlag = '', sel
...
+        const firstLength = a.data?.['Planting and Growth Windows']?.['Reliable Establishment']?.values.length;
+        const secondLength = b.data?.['Planting and Growth Windows']?.['Reliable Establishment']?.values.length;
+        console.log('length a, b', firstLength, secondLength);
+        if (firstLength && secondLength) {
+          firstDate = new Date(a.data?.['Planting and Growth Windows']?.['Reliable Establishment']?.values[firstLength - 1].split(' - ')[1]).toLocaleDateString('en-GB').split('/').reverse()
+            .join('');
+          secondDate = new Date(b.data?.['Planting and Growth Windows']?.['Reliable Establishment']?.values[secondLength - 1].split(' - ')[1]).toLocaleDateString('en-GB').split('/').reverse()
+            .join('');
+          console.log('firstDate ', a.label, firstDate);
+          console.log('secondDate ', b.label, secondDate);
```
[the remark this answers is not in the document]
Comment (Adamws33) [reply src/shared/constants.js]: deleted in commit 01f8066

File: src/pages/Landing/Landing.js (hunk 1 above)
Comment (Adamws33) [reply src/pages/Landing/Landing.js:203]: deleted in commit 01f8066

**Review by mikahpinegar — APPROVED** [review]

**Pull request merged**